### ISSUE TITLE
Fails when trying to pipe from command line

### ISSUE THREAD

**Issue by WhySoScherious** [jacobbednarz/csp-validator#2]
When trying to pipe from command line, I receive the following error:
./validate-csp:36:in `<main>': undefined method`match' for nil:NilClass (NoMethodError)

**Comment by jacobbednarz** [comment]
My guess is that `ARGV` can't be used for stdin. Might need to look at `ARGF` instead.

**Comment by jacobbednarz** [comment]
Could you give me an example of what you are trying to do so I can better assess how we can fix it?

**Comment by WhySoScherious** [comment]
Yes, that is correct, I have working code I can fork and do a pull request. Easy fix

diff --git a/validate-csp b/validate-csp
index ba7e3f0..9452832 100755
--- a/validate-csp
+++ b/validate-csp
@@ -33,7 +33,9 @@ OptionParser.new do |opts|
 end.parse!

 # Determine whether we are working a URL or a string.
-if ARGV[0].match(/^http/)
+if ARGV.length == 0
-    csp_data = ARGF.read
  +elsif ARGV[0].match(/^http/)
  response = Net::HTTP.get_response(URI(ARGV[0]))
  header = (options[:report_only]) ? response['Content-Security-Policy-Report-Only'] : response['Content-Security-Policy']

**Comment by WhySoScherious** [comment]
$ cat csp-header
default-src 'none'; script-src 'self' ssl.google-analytics.com 'sha256-xzi4zkCjuC8lZcD2UmnqDG0vurmq12W/XKM5Vd0+MlQ='; style-src 'self' maxcdn.bootstrapcdn.com fonts.googleapis.com; font-src fonts.gstatic.com maxcdn.bootstrapcdn.com; img-src 'self' ssl.google-analytics.com;

$ cat csp-header | ./validate-csp
./validate-csp:36:in `<main>': undefined method`match' for nil:NilClass (NoMethodError)

**Comment by jacobbednarz** [comment]
> Yes, that is correct, I have working code I can fork and do a pull request. Easy fix

Sounds great! Send over a PR and I'll check it out.

**Comment by ladykraken** [comment]
#please close
